google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/facet/impl/AppEngineJavaeeSupportContributor.java

```diff
@@ -68,7 +68,7 @@ private void setWebXmlServletVersion(Module module) {
     StartupManager.getInstance(module.getProject()).runWhenProjectIsInitialized(() -> {
       for (WebFacet webFacet : WebFacet.getInstances(module)) {
         final WebApp webApp = webFacet.getRoot();
-        if (webApp == null) {
+        if (webApp == null || !webApp.getXmlTag().isWritable()) {
```

Comment:
I guess its enough just to check if the XmlTag is writable like you are doing since each component will either be consistently writable or not based on the document, right?